### PR TITLE
feat: add additional tags/fields to pod metrics

### PR DESCRIPTION
**Pull request by rawkode**
Add state reason, which can eventually deprecated `terminated_reason`, to pod_container metrics; as well as a tag to represent the readiness of the container.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by rawkode** [comment]
I've removed `terminated_status` from the `README.md`, but it is still omitted from the plugin.

My thoughts were that we can hide / discourage it's use from the README and deprecate in favour of `state_reason` later.

Would like feedback

File: plugins/inputs/kube_inventory/pod.go [hunk 1]
```diff
@@ -37,30 +37,42 @@ func (ki *KubernetesInventory) gatherPod(p v1.Pod, acc telegraf.Accumulator) err
...
+		stateReason = cs.State.Terminated.GetReason()
 	case cs.State.Waiting != nil:
 		stateCode = 2
 		state = "waiting"
+		stateReason = cs.State.Waiting.GetReason()
+	}
+
+	ready := "unready"
+	if cs.GetReady() {
+		ready = "ready"
```
Comment (danielnelson) [inline plugins/inputs/kube_inventory/pod.go:59]: Consider naming this `readiness`, might flow better than `read=unready`

File: plugins/inputs/kube_inventory/pod.go [hunk 2]
```diff
@@ -37,30 +37,42 @@ func (ki *KubernetesInventory) gatherPod(p v1.Pod, acc telegraf.Accumulator) err
 
 func gatherPodContainer(nodeName string, p v1.Pod, cs v1.ContainerStatus, c v1.Container, acc telegraf.Accumulator) {
 	stateCode := 3
+	stateReason := ""
```
Comment (danielnelson) [inline plugins/inputs/kube_inventory/pod.go:40]: When this string is empty don't add it to the fields.

**Comment by rawkode** [comment]
Done. Thank you, @danielnelson

**Comment by rawkode** [comment]
Thanks, @danielnelson

**Pull request merged**